### PR TITLE
HPCC-14681 Add new updateLog method to ESP logging manager

### PR DESCRIPTION
**Pull request by wangkx**
1. The new updateLog method allows the updateLog request wth two
parameters. One is in IPropertyTree* format and the other is in
IInterface* format. 2. Support NoResend option.

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [x] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [x] Scalability
  - [x] Performance
  - [x] Security
  - [x] Thread-safety
  - [x] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browers
  - [ ] The component(s) render as expected

## Testing:

Test using the code for WU Billing which is under development.

### CODE REVIEW

**Comment by wangkx** [comment]
@afishbeck please review.

File: esp/logging/loggingmanager/loggingmanager.cpp [hunk 1]
```diff
@@ -106,6 +106,29 @@ bool CLoggingManager::updateLog(IEspContext& espContext, const char* option, con
     return bRet;
 }
 
+bool CLoggingManager::updateLog(const char* option, IPropertyTree* logInfo, IInterface* extraLog, StringBuffer& status)
```
Comment (afishbeck) [inline esp/logging/loggingmanager/loggingmanager.cpp]: The overloaded interface methods get very complicated very quickly.  Seems like a two part calling model would be better:
creating the log object, setting whatever available parms, and then calling a single updateLog method.
```
IEspLogEntry *entry = logmgr->createLogEntry();
entry->setEspContext(ctx);
entry->setOption(option);
entry->setContent(content);
logmgr->updateLog(entry, status);
```
I might be missing something, but even if we can't change the interface, couldn't these functions be commoned up at least internally?

For example something like:

```
bool updateLogImpl(IEspContext* espContext, const char* option, const char* logContent, StringBuffer& status)
{
  .... real functionality...
}
  
bool updateLog(IEspContext& espContext, const char* option, const char* logContent, StringBuffer& status)
{
    return updateLogImpl(&espContext, option, logContent, status);
}

bool updateLog(IEspContext& espContext, const char* option, const char* logContent, StringBuffer& status)
{
    return updateLogImpl(NULL, option, logContent, status);
}
```

Comment (wangkx) [inline esp/logging/loggingmanager/loggingmanager.cpp]: Good idea. I will implement that.

**Comment by wangkx** [comment]
@afishbeck please review my changes.

File: esp/logging/loggingmanager/loggingmanager.h [hunk 2]
```diff
@@ -25,13 +25,37 @@
...
+    virtual const char* getLogDatasets() = 0;
+};
+
 interface ILoggingManager : implements IInterface
 {
     virtual bool init(IPropertyTree* loggingConfig, const char* service) = 0;
-    virtual bool updateLog(IEspContext& espContext, const char* option, const char* logContent, StringBuffer& status) = 0;
-    virtual bool updateLog(const char* option, IEspContext& espContext, IPropertyTree* userContext, IPropertyTree* userRequest,
-        const char* backEndResp, const char* userResp, const char* logDatasets, StringBuffer& status) = 0;
-    virtual bool updateLog(IEspContext& espContext, IEspUpdateLogRequestWrap& req, IEspUpdateLogResponse& resp) = 0;
```
Comment (afishbeck) [inline esp/logging/loggingmanager/loggingmanager.h]: Is there existing code using this interface that has to change?

Comment (wangkx) [inline esp/logging/loggingmanager/loggingmanager.h]: No.

Comment (wangkx) [inline esp/logging/loggingmanager/loggingmanager.h]: I should say that D-ESDL is the only place which uses this interface right now. I did change that place in this revision.

Comment (afishbeck) [inline esp/logging/loggingmanager/loggingmanager.h]: Ok I see that DESDL change, if that's the only one, it's good.

**Comment by afishbeck** [comment]
@wangkx looks good.

**Comment by wangkx** [comment]
@richardkchapman This PR is ready to be merged.

**Pull request merged**